File: eprofiler-core/src/main/java/cz/encircled/eprofiler/asm/ClassAdapter.java
package cz.encircled.eprofiler.asm;

import cz.encircled.eprofiler.Util;
import org.objectweb.asm.ClassVisitor;
import org.objectweb.asm.MethodVisitor;

/**
 * @author Vlad on 23-May-16.
 */
public class ClassAdapter extends ClassVisitor {

    private final String owner;

    public ClassAdapter(ClassVisitor cv, final String owner) {
        super(Util.version, cv);
        this.owner = owner;
    }

    @Override
    public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
        MethodVisitor mv = cv.visitMethod(access, name, desc, signature, exceptions);
        if (mv != null && !name.equals("<init>") && !name.equals("<clinit>")) { // TODO configuration?
            mv = new ProfilerMethodAdapter(mv, access, name, desc, owner);
        }
        return mv;
    }

}

File: eprofiler-core/src/main/java/cz/encircled/eprofiler/asm/ClassAdapter.kt
package cz.encircled.eprofiler.asm

import cz.encircled.eprofiler.Util
import org.objectweb.asm.ClassVisitor
import org.objectweb.asm.MethodVisitor

/**
 * @author Kisel on 30.05.2016.
 */

class ClassAdapter(cv: ClassVisitor?, owner: String) : ClassVisitor(Util.version, cv) {

    override fun visitMethod(access: Int, name: String, desc: String, signature: String, exceptions: Array<String>): MethodVisitor? {
        var mv: MethodVisitor? = cv.visitMethod(access, name, desc, signature, exceptions)
        if (mv != null) {
            mv = ProfilerMethodAdapter(mv, access, name, desc)
        }
        return mv
    }

}